Fix for Issue #1 and some other enhancements

tprochazka opened the pull request:
I did this changes:
- fixed bug #1 (Editor return original editor instead secured one in fluent API
- added support for setStringSet() and getStringSet() method introduced in API 11 but in compatible way with API 8 (own implementation)
- added support for apply() introduced in API 9 with workaround for API 8

tprochazka committed:
- fix bug #1 (Editor return original editor instead secured one in fluent API
+ add support for setStringSet() and getStringSet() method introduced in API 11 but in compatible way with API 8 (own implementation)
+ add support for apply() introduced in API 9 with workaround for API 8

tprochazka committed:
- fix Android 2.2 (API 8 compatibility)

kovmarci86 commented on pom.xml:
```diff
@@ -31,15 +31,25 @@
+	<repository>
+		<id>nexus</id>
+		<name>Inmite 3rd party Repo</name>
+		<url>http://mvn.office.inmite.eu/nexus/content/repositories/thirdparty</url>
```
mvn.office.inmite.eu is not reachable from outside

kovmarci86 commented:
Hi tprochazka,
Thank you for the pull-request! I'm happy to see someone's contribution.
The most of the changes are reasonable and great, but there are some issues.
The repositories are not reachable from outside of your company therefore it must not be there.
For the SecuredEditor's builder method return statements: you were correct, it returned the wrong editor, but I'm not sure it is good to return the SecuredEditor, as the main concept is to give completely the same interface than SharedPreferences. The SecuredEditor is a public class therefore it must not contain any extra public methods to avoid any kind of dependency to this library. (it can be plugged in easily, but also can be removed if someone decides not to use it)
This is the case for the edit method of SecureSharedPreferences class.
What do you think about this?
Thanks again, best regards,
Marcell

tprochazka committed:
- revert <distributionManagement> to original state

tprochazka commented:
I'm sorry, repository was added by mistake. I removed it.

I also agree with you second objection, but there is not possible because I added support for methods which was introduced in newer version that API 8, like apply() or setStringSet(). These methods are implemented to work also on API 8. So you can write the same code that will work on API 8 and API 17. If SecureEditor will return only Editor it will not work on old Android API because some methods are missing  here. You can try it.

tprochazka commented:
So, what do you mean?
